app/models/cart.rb

```diff
@@ -0,0 +1,4 @@
+class Cart < ApplicationRecord
+  has_many :line_items
+  has_many :products, :through => :line_items
```

Comment:
has_many :products, through: :line_items
 
 has_many :products, :through => :line_items    this is old convention